### ISSUE TITLE
Included files sometimes not escaped

### ISSUE THREAD

**Issue by jakubheglas** [mslinn/jekyll_flexible_include_plugin#7]
I am running into a truly weird issue where sometimes the same included files are simply not escaped and sometimes they are, without any apparent connection, despite having explicitly set `do_not_escape`.

## Implementation

1. I have couple similar pages which all reference the same include (standard Jekyll file within `_includes` folder)
2. This include references a relatively basic SVG file from `assets/images` folder to be rendered on page using `flexible_include`

## Setup 
So far, these two pages render unescaped SVG nicely:
- index.md
- about.md

Other pages show escaped markup instead:
- team.md
- any other page directly based off (duplicated) of two functional pages above

Do note, as I'm in relatively early stages of development, pages about.md and team.md are entirely identical, as I basically stopped any work scratching my head what's going on and thus not proceeding.

## Observations
Above is of course just to illustrate the weirdness of the issue. Following (random) observations have been made:

- [x] Implementation is as basic as it can get, I don't believe I'm trying to do anything remotely exotic.
- [x] I tried to clear Jekyll cache to no avail. 
- [x] I tried to deploy to Netlify with identical result, so I don't believe there's any other potential cache issue involved.
- [x] Had to add `key-value-parser` into my Gemfile in order to make the plugin work.
- [x] My ruby version is 2.7.4 and I also use node v14.20.0 on the side to compile Tailwind

Is there any sort of trace or any other action I could take to help figure this out?

Other than this, great plugin, cheers!

**Comment by mslinn** [comment]
Please show some of the statements you are using for the includes. Also, what version are you using?

**Comment by jakubheglas** [comment]
So the setup as per above looks like this:

**Team page** has pretty much nothing but this include
`{% include hero.html title="Team" %}`

**The _hero_ partial** among other fairly basic html has this flexible_include
`{% flexible_include assets/images/heroMaskLines.svg do_not_escape %}`

The version is, as README recommends, 2.0.0.

**Comment by mslinn** [comment]
I wonder if the reason the included contents are sometimes not escaped is that your Jekyll processes run on different VMs or file systems, with different permissions or file versions.

You could update the plugin to the current version (v2.0.13), but I do not expect that to affect the problem, however.
```shell
$ bundle update
```

BTW, the installation instructions cause v2.0.0 or a compatible later version to be installed. The way to check the version is like so:
```shell
$ gem info jekyll_flexible_include
```

You could enable the plugin's logging, or crank up its log level. See [Custom Logging in Jekyll Plugins](https://www.mslinn.com/jekyll/4100-custom-logging-in-jekyll-plugins.html) and [Jekyll Plugin Template Collection](https://www.mslinn.com/jekyll/10400-jekyll-plugin-template-collection.html)

You can run the plugin under a debugger, as described in [Debugging Jekyll Plugins with an IDE](https://www.mslinn.com/jekyll/6600-jekyll-debugging.html). It is usually best to attach to the Jekyll process, instead of launching under a debugger. To do that:

1. Read through the article
2. Install the debugging gems
3. Clone this repo
4. Change the gem reference in your Jekyll project from this:
   ```
   gem 'jekyll_flexible_include', '~> 2.0.0'
   ```
   to this:
   ```
   gem 'jekyll_flexible_include', ['~> 2.0.0'](path: "/path/to/jekyll_flexible_include_plugin_directory")
   ```
6. Launch vscode
7. Set breakpoints in the flexible_include plugin
8. Run `bin/attach .` from the cloned repo.

**Comment by mslinn** [comment]
I just released v2.0.14, which defines `key-value-parser` as a dependency. Thanks for mentioning it.
